### PR TITLE
(PC-6142) Display the right non-beneficiary header based on the user age

### PR DESCRIPTION
**Pull request by blixit**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-6142

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [x] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [x] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*- 18 y | 18 y | + 18 y | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
| image  | image | image |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/profile/components/NonBeneficiaryHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { _ } from 'libs/i18n'
+import { HeaderBackground } from 'ui/svg/HeaderBackground'
+import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'
+
+import { computeCreditExpiracy } from '../utils'
+
+import { YoungerBadge } from './YoungerBadge'
+
+interface NonBeneficiaryHeaderProps {
+  email: string
```
Comment (AnoukHello) [inline src/features/profile/components/NonBeneficiaryHeader.tsx:16]: Pourquoi on a besoin de l'email de l'utilisateur dans ce composant ?

File: src/features/profile/utils.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Formats an iso date to a slashed french date.
+ * @param ISOBirthday the birthday date into the ISO 8601 format %Y-%m-%dT%H:%M:%S
+ */
+export function computeCreditExpiracy(ISOBirthday: string) {
```
Comment (AnoukHello) [inline src/features/profile/utils.ts:5]: Que penses-tu de renommer cette fonction en `computeEligibilityExpiracy` ?

File: src/features/profile/components/NonBeneficiaryHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  let body = null
+  switch (true) {
+    case age > 18:
+      body = null
+      break
+    case age === 18:
+      body = (
+        <View testID="18-view">
+          <Typo.Caption>{_(t`Tu es elligible jusqu'au\u00a0`) + expiracyDate}</Typo.Caption>
```
Comment (AnoukHello) [inline src/features/profile/components/NonBeneficiaryHeader.tsx:34]:
```suggestion
          <Typo.Caption>{_(t`Tu es éligible jusqu'au\u00a0`) + expiracyDate}</Typo.Caption>
```

File: src/features/profile/components/NonBeneficiaryHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+const HeaderBackgroundWrapper = styled.View({
+  maxHeight: 64,
+  overflow: 'hidden',
+  position: 'relative',
+  alignItems: 'center',
+})
+
+const Title = styled(Typo.Title4)({
+  position: 'absolute',
+  top: 32,
```
Comment (AnoukHello) [inline src/features/profile/components/NonBeneficiaryHeader.tsx:73]: `getSpacing(8)` ?

File: src/features/profile/components/NonBeneficiaryHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+/** Add 1 pixel to avoid 1 white pixel on androids */
+const screenWidth = Dimensions.get('window').width + 1
+
+const Container = styled.View({
+  flex: 1,
+})
+
+const HeaderBackgroundWrapper = styled.View({
+  maxHeight: 64,
```
Comment (AnoukHello) [inline src/features/profile/components/NonBeneficiaryHeader.tsx:65]: `getSpacing(16)` ?

File: src/features/profile/components/NonBeneficiaryHeader.tsx (hunk 1 above)
Comment (blixit) [reply src/features/profile/components/NonBeneficiaryHeader.tsx:16]: pour la redirection vers IDCheck. j'ai supprimé le code qui en avait besoin, mais la prop sera valide quand je reprendrai le ticket 6551

File: src/features/profile/components/NonBeneficiaryHeader.tsx (hunk 3 above)
Comment (blixit) [reply src/features/profile/components/NonBeneficiaryHeader.tsx:34]: ah le zeplin m'a bêtement induit en erreur. Thanks !!

**Pull request merged**